diesel_derives/src/non_aggregate.rs

```diff
@@ -0,0 +1,32 @@
+use proc_macro2::{self, Span, Ident};
+use syn;
+
+use util::*;
+
+pub fn derive(item: syn::DeriveInput) -> Result<proc_macro2::TokenStream, Diagnostic> {
+    let mut generics = item.generics.clone();
+
+    {
+        let where_clause = generics.make_where_clause();
+        for type_param in item.generics.type_params() {
+            where_clause.predicates.push(parse_quote!(#type_param: NonAggregate));
```

Comment:
Several of the types that now derive `NonAggregate` had `Self: Expression` on their impls before. Not sure if that matters?

Reply:
It's just from back when `Expression` was a supertrait of `NonAggregate`. It's safe to remove.